src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -141,6 +143,7 @@ public void setShouldStartLine(boolean flag) {
             TokenTypes.LITERAL_FINALLY,
             TokenTypes.LITERAL_IF,
             TokenTypes.LITERAL_ELSE,
+            TokenTypes.LAMBDA,
```

Comment:
Good catch, as are not ready to provide users good support for lambda (anonymous classes) . It should not be a default.

Reference to #4177 and #4178